core/src/main/java/com/linecorp/armeria/internal/annotation/AnnotatedHttpServiceFactory.java

```diff
@@ -290,6 +375,21 @@ private static AnnotatedHttpServiceElement create(String pathPrefix, Object obje
                                   "invalid HTTP status code: %s (expected: >= 0)", statusCode);
                     return HttpStatus.valueOf(statusCode);
                 });
+        final HttpHeaders defaultHeaders = HttpHeaders.of(defaultResponseStatus(defaultResponseStatus, method));
+
+        final HttpHeaders defaultTrailingHeaders = HttpHeaders.of();
+
+        setAdditionalHeader(defaultHeaders, clazz, "header", "class", AdditionalHeader.class);
+        setAdditionalHeader(defaultHeaders, method, "header", "method", AdditionalHeader.class);
+        setAdditionalHeader(defaultTrailingHeaders, clazz, "trailer", "class", AdditionalTrailer.class);
+        setAdditionalHeader(defaultTrailingHeaders, method, "trailer", "method", AdditionalTrailer.class);
+
+        if (ArmeriaHttpUtil.isContentAlwaysEmpty(defaultHeaders.status()) && !defaultTrailingHeaders
+                .isEmpty()) {
+            logger.warn("The content for HTTP status ({}) should be always empty. " +
+                        "Trailing headers for the method '{}' might be ignored.",
```

Comment:
- `A response with HTTP status code '{}' cannot have a content. Trailing headers defined at '{}' will be ignored.`
- How about also mentioning the class name? e.g. in the format of `<className>.<methodName>()`